test/flow.js

```diff
@@ -34,19 +27,99 @@ const mutation = gql`
 
 type IQuery = {
   foo: string,
+  bar: string,
 };
 
-// common errors
-
 const withData: OperationComponent<IQuery> = graphql(query);
 
-const ComponentWithData = withData(({ data: { foo } }) => {
-  // $ExpectError
+// Test a functional component
+const FunctionalWithData = withData(({ data: { foo } }) => {
+  // $ExpectError string type beeing treated as numerical
   if (foo > 1) return <span />;
 
   return null;
 });
 
+// Test class component, this requieres a stricter definition
+type BasicComponentProps = {
+  data: IQuery & QueryProps,
```

Comment:
You could provide some utility type like `GraphQLProps<IQuery>` and recommend users spread that in their props type:

```
type Props = {
  ... GraphQLProps<IQuery>,
  myProp: number,
};
```

This also means you could include `mutate` as mixed. For mutations you could have a different type like `GraphQLMutationProps<IMutation>`.

Reply:
Tried, can't get it to work :-(

Reply:
@calebmer would you be able to help out here? I won't be able to try the upgrade today